DataHolderLib/Project.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "Project.h"
+
+#include <algorithm>
+
+// ThirdParty/logog
+#include <logog/include/logog.hpp>
+
+#include "BaseLib/FileTools.h"
+#include "BaseLib/uniqueInsert.h"
+
+#include "MeshLib/Mesh.h"
+
+namespace DataHolderLib
+{
+
+Project::~Project()
+{
+	for (MeshLib::Mesh* m : _mesh_vec)
+		delete m;
+}
+
+void Project::addMesh(MeshLib::Mesh* mesh)
+{
+	std::string name = mesh->getName();
+	getUniqueName(name);
+	mesh->setName(name);
+	_mesh_vec.push_back(mesh);
+}
+
+std::vector<MeshLib::Mesh*>::const_iterator Project::findMeshByName(
+		std::string const& name) const
+{
+	return const_cast<Project&>(*this).findMeshByName(name);
```

Comment:
again, this was not done by me, it's straight from projectdata. i do need the non-const method. having both is no bother so i'm just keeping it.